### PR TITLE
[FIRRTL] add WireDFT pass 

### PR DESCRIPTION
**Pull request by youngar**
This adds a pass to wire the "test_en" port of all modules named
EICG_Wrapper to a signal marked by the "DFTEnable" annotation.  This is
used to support Design For Testing clock gating.

### CODE REVIEW

**Comment by lattner** [comment]
Thank you for splitting the independent pieces out into their own patch.  I did a quick skim over the other code and this looks reasonable to me.  It would be good for someone to do a detailed review tho

**Comment by youngar** [comment]
Prereqs have been merged

File: include/circt/Dialect/FIRRTL/Passes.td [hunk 1]
```diff
@@ -114,6 +114,19 @@ def CreateSiFiveMetadata : Pass<"firrtl-emit-metadata", "firrtl::CircuitOp"> {
   ];
 }
 
+def WireDFT : Pass<"firrtl-dft", "firrtl::CircuitOp"> {
+  let summary = "Wires test enables to clock gates for DFT infrastructure";
+  let description = [{
+    This pass will take a 1-bit signal targeted by
+    `DFTTestModeEnableAnnotation` and wires it to the `test_en` port of every
+    module named `EICG_wrapper`. This will create ports in any intermediate
```
Comment (darthscsi) [inline include/circt/Dialect/FIRRTL/Passes.td:122]: This description makes me sad.  (not the text, the text is fine, but the functionality)

**Review by darthscsi — APPROVED** [review]
LGTM

**Comment by youngar** [comment]
rebased for safety, looks like we're still good 👍

**Pull request merged**